Add AttributeTable module

JakobMiksch opened the pull request:
This PR adds a attribute table to Wegue.

- it can be activated in the configuration file like other modules
- it provides a drop-down-menu with all vector layers and displays them accordingly in the table
- works for both desktop and mobile (see GIFs below)
- a click on the column name sorts the table
- the human-readable names of the columns can be defined using the layer-property `columnMapping`
- the documentation has been updated

**Desktop**
![desktop](https://user-images.githubusercontent.com/15704517/111508988-90b55800-874c-11eb-9d82-740066ef8056.gif)

**Mobile**
![mobile](https://user-images.githubusercontent.com/15704517/111509176-bfcbc980-874c-11eb-9e58-f3aed97a682a.gif)

chrismayer commented on src/components/attributeTable/AttributeTableWin.vue:
```diff
@@ -0,0 +1,108 @@
+    bottom: 35px;
+}
+
+@media only screen and (max-width: 600px) {
```
Does `600px` correspond to the Vuetify breakpoint?

JakobMiksch replied:
`600px` is the Vuetify breakpoint for phones https://vuetifyjs.com/en/features/breakpoints/

chrismayer commented on src/components/attributeTable/ToggleButton.vue:
```diff
@@ -0,0 +1,52 @@
+    }
+  },
+  created () {
+    var me = this;
```
Not necessary since you are using the arrow functions below. Please use `this` instead.

JakobMiksch replied:
done

chrismayer commented on src/components/layerlist/ToggleButton.vue:
```diff
@@ -31,7 +31,6 @@ export default {
   created () {
     var me = this;
     // TODO move to a father class
-    console.log('on');
```
Please remove

JakobMiksch replied:
this is already removed

chrismayer commented on src/components/layerlist/ToggleButton.vue:
```diff
@@ -41,7 +40,6 @@ export default {
   },
   methods: {
     toggleUi () {
-      console.log(this.win);
```
Please remove

JakobMiksch replied:
this is already removed

chrismayer reviewed (commented):
Thanks for this @JakobMiksch ! Very cool addition. Please address the comments I added either for clarification or code polish.

JakobMiksch commented on package.json:
```diff
@@ -13,6 +13,7 @@
     "start": "node build/dev-server.js",
     "build": "node build/build.js",
     "unit": "cross-env BABEL_ENV=test karma start test/unit/karma.conf.js --single-run",
+    "lint-fix": "eslint --fix --ext .js,.vue src app test/unit/specs",
```
true, I move it to another PR

JakobMiksch replied:
and also removed it from this PR

JakobMiksch commented on src/components/attributeTable/AttributeTable.vue:
```diff
@@ -0,0 +1,135 @@
+    :loading-text="loadingText"
+    :headers="headers"
+    :items="records"
+     mobile-breakpoint="0"
```
I added a comment.

JakobMiksch commented on src/components/attributeTable/AttributeTable.vue:
```diff
@@ -0,0 +1,135 @@
+    }
+  },
+  methods: {
+    populateTable () {
```
done

JakobMiksch commented on src/components/attributeTable/AttributeTable.vue:
```diff
@@ -0,0 +1,135 @@
+
+      let headers = [];
+      if (this.layer.get('columnMapping')) {
+        for (const [propertyName, DisplayName] of Object.entries(this.layer.get('columnMapping'))) {
```
done

JakobMiksch commented on src/components/attributeTable/AttributeTable.vue:
```diff
@@ -0,0 +1,135 @@
+        this.layer.setVisible(true);
+      }
+    },
+    applyRecordsFromOlLayer () {
```
done

JakobMiksch commented on src/components/attributeTable/AttributeTable.vue:
```diff
@@ -0,0 +1,135 @@
+        //       --> it might be better to check if the
+        //           type of the property is valid for the table
+        const filtered = keys.filter(
+          key => ((key !== 'geometry') && (key !== 'the_geom'))
```
I added it as a property to the `AttributeTable` component

JakobMiksch commented:
@chrismayer thanks for the  review. I addressed your comments. 

Additionally I added basic tests for `AttributeTable`, `AttributeTableWin`, `ToggleButton` (Attribute Table). 

And I added a label to the drop-down menu. See gif below

![Peek 2021-03-24 11-33](https://user-images.githubusercontent.com/15704517/112296045-cdc1a300-8c94-11eb-8f2f-d327112a3efd.gif)

chrismayer commented on src/components/attributeTable/AttributeTableWin.vue:
```diff
@@ -0,0 +1,114 @@
+      >{{title}}</v-toolbar-title>
+      <v-spacer></v-spacer>
+      <v-select
+          v-model="selectedItem"
```
Indentation

JakobMiksch replied:
done with a562077

chrismayer commented on src/components/attributeTable/AttributeTableWin.vue:
```diff
@@ -0,0 +1,114 @@
+  </v-toolbar>
+
+  <wgu-attributetable
+  v-if="layerId"
```
Indentation

JakobMiksch replied:
done with a562077

chrismayer commented on src/components/attributeTable/AttributeTableWin.vue:
```diff
@@ -0,0 +1,114 @@
+      const mapLayers = this.map.getLayers();
+      mapLayers.forEach(layer => {
+        if (layer instanceof VectorLayer &&
+            layer.get('name') !== 'Measure Layer') {
```
Please use the `lid` of the measure layer instead of the `name`. Once we have i18n this approach won't work anymore.

JakobMiksch replied:
done with 3bed90c

chrismayer reviewed (commented):
Thanks again. Several comments. The one with the detecting the measure layer should be addressed before merging IMHO.

JakobMiksch commented:
@chrismayer I addressed your comments

JakobMiksch commented:
@chrismayer I noticed that I did not copy the files from `app` to `appStarter`. I fixed it with 
0170b8c. I also updated the branch to the latest `master`

chrismayer commented:
Thank you very much for this @JakobMiksch , LGTM now! 'Ill merge...
